File: 0890-lemonade-change/0890-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int five = 0;
        int ten = 0;
        int twenty = 0;
        for(int i=0;i<bills.length;++i){
            int x = bills[i];
            if(x==5) five++;
            if(x==10){
                if(five==0) return false;
                five--;
                ten++;
            }
            if(x==20){
                if(ten==0){
                    if(five<3) return false;
                    else{
                        five=five-3;
                    }
                }
                if(ten>0){
                    if(five==0)return false;
                    if(five>0){
                        ten--;
                        five--;
                    }
                }
            }
        }
        return true;
    }
}

File: 0890-lemonade-change/0890-lemonade-change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        ios_base::sync_with_stdio(false); cin.tie(NULL);
        int fives = 0, tens = 0;
        for (const int &x: bills) {
            if (x == 5) fives++;
            else if (x == 10) {
                if (fives > 0) fives--, tens++;
                else return false;
            } else if (x == 20) {
                if (tens > 0 && fives > 0) tens--, fives--;
                else if (fives >= 3) fives -= 3;
                else return false;
            }
        }
        return true;
    }
};